### DISCUSSION TITLE
图例一旦超过上百个,图表加载会非常卡顿

### DISCUSSION THREAD

**Discussion by qtj4js** [antvis/G2#6186] [Q&A]
![1edcce7c895b5f1c7e4cdf7025581dd3](https://github.com/antvis/G2/assets/47909978/cd72df1c-d280-44fd-be3a-ea3710399351)
我们的需求一般是有一个非常多数据的图表,然后通过外部我们自己的筛选获取想要的数据
但是现在这个原始图表加载就非常卡顿,定位到是图例加载非常的慢.
这个有什么办法解决优化么?
看G2的源码没有找到任何可配置优化的项

**Comment by hustcc** [comment]
1. 图例分页
2. 对于饼图，显示几百个区块已经没有什么意义了，所以可以从数据处理上，把超过 10 块的数据合并成「其他」，这个也是饼图设计上推荐的做法。

**Reply by pearmini** [reply]
目前内部的实现比较耗性能，所以可以自己实现高性能的版本：https://g2.antv.antgroup.com/manual/extra-topics/customization#%E8%87%AA%E5%AE%9A%E4%B9%89%E5%9B%BE%E4%BE%8Blegend

**Reply by qtj4js** [reply]
并不只是饼图,所有图表都会遇到这个情况,图例一多久非常卡顿.
这里只是以饼图举个例子
可以考虑一下是不是优化
